### PR TITLE
JENKINS-49929 Regression: Log fails to auto update in karaoke mode

### PR DESCRIPTION
**Pull request by sophistifunk**
# Description

See [JENKINS-49929](https://issues.jenkins-ci.org/browse/JENKINS-49929).

# Submitter checklist
- [X] Link to JIRA ticket in description, if appropriate.
- [X] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by sophistifunk** [comment]
I haven't the foggiest idea how we would test this mechanically, only a certainly that writing a reliable ATH case for it would take 2 weeks.

**Comment by kzantow** [comment]
Prettier is _not_ fun for changes like this, hard to see what the actual issue was, needs to be run against the whole codebase in one fell swoop and then applied during commit time as well

File: blueocean-dashboard/src/main/js/components/karaoke/components/Step.jsx [hunk 1]
```diff
@@ -117,26 +122,27 @@ export class Step extends Component {
...
-                location,
-                hasMore,
-                scrollToBottom,
-                logArray,
-                currentLogUrl,
-                key: step.logUrl,
-                prefix: `step-${step.id}-`,
-            }}
-            />);
+            children = (
```
Comment (kzantow) [inline blueocean-dashboard/src/main/js/components/karaoke/components/Step.jsx]: oh thank God that spread syntax is gone

File: blueocean-dashboard/src/main/js/components/karaoke/components/Step.jsx [hunk 2]
```diff
@@ -117,26 +122,27 @@ export class Step extends Component {
...
+            );
         } else if (step.isInputStep) {
             children = <InputStep step={step} key="step" classicInputUrl={classicInputUrl} />;
         } else if (!logArray && step.hasLogs) {
             children = <span key={'span'}>&nbsp;</span>;
         }
         const getLogForNode = () => {
             if (this.pager.log === undefined || (this.pager.log && !this.pager.log.data)) {
-                const cfg = { url: step.logUrl };
+                const cfg = { url: step.logUrl, followAlong: this.props.augmenter.karaoke };
```
Comment (kzantow) [inline blueocean-dashboard/src/main/js/components/karaoke/components/Step.jsx]: ... wheeeeeeee karaoke!

File: blueocean-dashboard/src/main/js/components/karaoke/components/Step.jsx (hunk 1 above)
Comment (sophistifunk) [reply blueocean-dashboard/src/main/js/components/karaoke/components/Step.jsx]: Word. I kills it where I finds it.

File: blueocean-dashboard/src/main/js/components/karaoke/components/Step.jsx (hunk 2 above)
Comment (sophistifunk) [reply blueocean-dashboard/src/main/js/components/karaoke/components/Step.jsx]: Yes, the other changes I fixed for readability while trying to understand the issue. Figured no reason to throw them away and have to do it again next week.

**Pull request merged**